environments/devshift-cluster.deploy.sh

```diff
@@ -18,24 +18,12 @@ export PROXIED_K8S_API_SERVER="${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_P
 # This is our proxy that we will connect to
 #export K8S_API_SERVER="" # Intentionally left blank, we want to use the support for detecting the URL
 
-if [ -z "${OAUTH_ISSUER}" ]; then
-  export OAUTH_ISSUER="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"
-fi
-if [ -z "${OAUTH_SCOPE}" ]; then
-  export OAUTH_SCOPE="user:full"
-fi
-if [ -z "${OAUTH_CLIENT_ID}" ]; then
-  export OAUTH_CLIENT_ID="fabric8"
-fi
-if [ -z "${K8S_API_SERVER_PROTOCOL}" ]; then
-  export K8S_API_SERVER_PROTOCOL="http"
-fi
-if [ -z "${K8S_API_SERVER_BASE_PATH}" ]; then
-  export K8S_API_SERVER_BASE_PATH="/_p/oso"
-fi
-if [ -z "${WS_K8S_API_SERVER}" ]; then
-  export WS_K8S_API_SERVER=${PROXIED_K8S_API_SERVER}
-fi
+export OAUTH_ISSUER=${OAUTH_ISSUER:-"https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"}
```

Comment:
Why remove the IF check? What does that buy you? Isn't it there so that you only export things you have?

Reply:
@joshuawilson 

>  Isn't it there so that you only export things you have?

No, the `-z` checks if a value is not set (is zero). If something is zero, then we set it to a default. And for this there's a shortcut in bash that goes like this:

```sh
foo=${blabla:-"bar"}
```

This sets the variable `foo` to the value `"bar"` but only if `blabla` wasn't set above (e.g. through an environment variable). I find this much shorter, readable, hence maintainable than the `if` construct. That's why I've made the change.